src/App.tsx

```diff
@@ -1,29 +1,288 @@
+   const handleBuyMaxClick = () => {
+      const repeatMax = () => {
+         timeIdRef.current = setTimeout(() => {
+            if (canIstillBuy()) {
+               console.log(canIstillBuy(), canIstillBuyRef.current, tickspeedPrice);
+               handleTickBtnClick();
+               repeatMax();
+            }
+         }, 20);
+
+         return () => clearTimeout(timeIdRef.current);
+      };
+      repeatMax();
+   };
+
+   // creates constantly an uptodate object which is then saved later
+   useEffect(() => {
+      dataRef.current = {
+         // ...dataRef.current,
+         antimatter,
+         tickspeedPrice,
+         firstDimCount,
+         firstDimFactor,
+         firstDimPrice,
+         firstDimFactorCount,
+         secondDimCount,
+         secondDimFactor,
+         secondDimPrice,
+         secondDimFactorCount,
+         thirdDimCount,
+         thirdDimFactor,
+         thirdDimPrice,
+         thirdDimFactorCount,
+         resetGameCounter,
+         highesDim,
+         timerIdRef,
+         idRef,
+         clockSpeedRef,
+      };
+      //console.log('dataRef object update cycle: ', dataRef.current ?? '');
+   });
+
+   // saves the created object every minuit to localStorage
+   useEffect(() => {
+      const startSave = () => {
+         timerIdRef.current = setTimeout(() => {
+            //localStorage.removeItem('dataRef');
+            localStorage.setItem('data', JSON.stringify(dataRef.current));
+            console.log('data save cycle: ', dataRef.current);
+            startSave();
+         }, 60000);
+      };
+      startSave();
+
+      // if we ever unmount / destroy this component instance, clear the timeout
+      return () => clearTimeout(timerIdRef.current);
+   }, []);
+
+   // this prints the saved object from local storage. It is just for dev. No final purpose
+   useEffect(() => {
+      const printStorage = () => {
+         idRef.current = setTimeout(() => {
+            let savedDataRef = JSON.parse(localStorage.getItem('data') ?? '');
+            console.log('dataRef from storage', savedDataRef);
+            printStorage();
+         }, 60000);
+      };
+      printStorage();
+
+      return () => clearTimeout(idRef.current);
+   }, []);
+
+   /* resets the game to unlock new dimension */
+   const handleResetGameClick = () => {
+      clockSpeedRef.current = 2000;
+      timerIdRef.current = -1;
+      idRef.current = -1;
+
+      setTickspeedPrice(10);
+      setAntimatter(1000);
+
+      setFirstDimFactor(1.1);
+      setFirstDimCount(0);
+      setFirstDimPrice(10);
+      setFirstDimFactorCount(0);
+
+      setSecondDimFactor(1.1);
+      setSecondDimCount(0);
+      setSecondDimPrice(100);
+      setSecondDimFactorCount(0);
+
+      setThirdDimFactor(1.1);
+      setThirdDimCount(0);
+      setThirdDimPrice(1000);
+      setThirdDimFactorCount(0);
+
+      // increase ResetCounter by one
+      setResetGameCounter(prev => prev + 1);
+   };
```

Comment:
Reasonable.  One thing you might improve here: try to have only one copy of these initial values, instead of duplicating the numbers used to initialize state near the top of this component.